src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala

```diff
@@ -81,35 +81,37 @@ class DataProcessor[F[_]]()(implicit me: MonadError[F, Throwable]) {
   }
 
   /** Returns the metadata fragment urls from the element
-    * @param elem
-    *   The element to search
+    * @param entityResponseElement
+    *   The EntityResponse element to search
     * @return
     *   A `Seq` of `String` wrapped in the F effect with the fragment URLS
     */
-  def fragmentUrls(elem: Elem): F[Seq[String]] = {
-    val fragments = elem \ "AdditionalInformation" \ "Metadata" \ "Fragment"
-    me.pure(fragments.map(_.text))
+  def fragmentUrls(entityResponseElement: Elem): F[Seq[String]] = {
+    val fragmentUrls = entityResponseElement \ "AdditionalInformation" \ "Metadata" \ "Fragment"
+    me.pure(fragmentUrls.map(_.text))
   }
 
   /** Returns a list of metadata content
-    * @param elems
-    *   The element which contains the content
+    * @param metadataResponseElements
+    *   The MetadataResponse elements that contains the fragments
     * @return
     *   A `Seq` of String representing the metadata XML or an error if none are found
     */
-  def fragments(elems: Seq[Elem]): F[Seq[String]] = {
-    val metadataObjects = elems.map { elem =>
-      val eachContent: NodeSeq = elem \ "MetadataContainer" \ "Content"
-      eachContent.flatMap(_.child).toString()
-    }
-    metadataObjects.filter(!_.isBlank) match {
-      case Nil =>
+  def fragments(metadataResponseElements: Seq[Elem]): F[Seq[String]] = {
```

Comment:
As we've talked about, you can simplify this by returning the `metadataResponseElements` in the exception and just using List instead of Maps. If it turns out this is annoying down the line, we can change it.